### ISSUE TITLE
[FNL-10] flushPage and VersionedSegment write-ahead logging are incompatible

### ISSUE THREAD

**Issue by dynamobi-build** [LucidDB/luciddb#498]
<b><i>[reporter="jvs", created="Wed, 18 Jan 2006 17:10:47 -0500 (GMT-05:00)"]</i></b>
flushPage doesn't honor the write-ahead logging protocol, so when we get down into the cache, we assert because the segment complains that the page isn't in a good state to be flushed.  Exposed by calls in LcsClusterNodeWriter.cpp.<br/>

**Comment by dynamobi-build** [comment]
<b><i>[author="jvs", created="Wed, 18 Jan 2006 17:11:30 -0500 (GMT-05:00)"]</i></b>
To reproduce, uncomment those calls and then run "ant txn" from LucidDB.<br/>

**Comment by dynamobi-build** [comment]
<b><i>[author="jvs", created="Sat, 21 Jan 2006 21:35:35 -0500 (GMT-05:00)"]</i></b>
Fixed and uncommented the calls.<br/>
<br/>
Well, fixed is a bit of an overstatement.  What I did was ignore the flush request in the case where the log page wasn't done yet.  Real fix would require some logic up at VersionedSegment level, interacting with SegPageLock.  But my fix is quite good enough for this particular case (only the flush request for the very first page in an incremental load will be ignored; all other pages are guaranteed to be new pages, so they don't need a write-ahead pre-image).
